datasets/ehealth_kd/ehealth_kd.py

```diff
@@ -0,0 +1,159 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""TODO: Add a description here."""
+
+from __future__ import absolute_import, division, print_function
+
+import datasets
+
+
+# TODO: Add BibTeX citation
+# Find for instance the citation on arxiv or on the dataset repo/website
+_CITATION = """\
+@InProceedings{huggingface:dataset,
+title = {A great new dataset},
+authors={huggingface, Inc.
+},
+year={2020}
+}
+"""
+
+# TODO: Add description of the dataset here
+# You can copy an official description
+_DESCRIPTION = """\
+This new dataset is designed to solve this great NLP task and is crafted with a lot of care.
+"""
+
+_HOMEPAGE = "https://knowledge-learning.github.io/ehealthkd-2020/"
+
+_LICENSE = ""
+
+_URL = "https://raw.githubusercontent.com/knowledge-learning/ehealthkd-2020/master/data/"
+_TRAIN_DIR = "training/"
+_DEV_DIR = "development/main/"
+_TEST_DIR = "testing/scenario1-main/"
+_TEXT_FILE = "scenario.txt"
+_ANNOTATIONS_FILE = "scenario.ann"
+
+
+class eHealthKD(datasets.GeneratorBasedBuilder):
+    """TODO: Short description of my dataset."""
+
+    VERSION = datasets.Version("1.1.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(name="ehealth_kd", version=VERSION, description="eHealth Knowledge Discovery dataset"),
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "sentence": datasets.Value("string"),
+                    "entities": [
+                        {
+                            "ent_id": datasets.Value("string"),
+                            "ent_text": datasets.Value("string"),
+                            "ent_tag": datasets.Value("string"),
+                            "start_character": datasets.Value("int32"),
+                            "end_character": datasets.Value("int32"),
+                        }
+                    ],
+                    "relations": [
+                        {
+                            "rel_id": datasets.Value("string"),
+                            "rel_tag": datasets.Value("string"),
+                            "arg1": datasets.Value("string"),
+                            "arg2": datasets.Value("string"),
+                        }
+                    ],
+                }
+            ),
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        urls_to_download = {
```

Comment:
if you wanted, you could save some duplicate code with something like

```python
urls_to_download = {k:[f"{_URL}{v}{_TEXT_FILE}", f"{_URL}{v}{_ANNOTATIONS_FILE}"] for k,v in zip(["train", "dev", "test"], [_TRAIN_DIR, _DEV_DIR, _TEST_DIR])}
```

Reply:
Looks really nice, thanks!